### ISSUE TITLE
Instrument Probability feature

### ISSUE THREAD

**Issue by firolightfog** [schismtracker/schismtracker#225]
Hello,
I'm wondering if adding probability to instruments would be a technically a challenging issue. It'd be clearly a great enhancement!

![image](https://user-images.githubusercontent.com/34127628/71571747-2a5b0000-2adc-11ea-88df-57247bae9ef0.png)

Meaning:
0% - the instrument is never triggered
50% - only half of the notes are played on avg.
100% - the instrument always triggered if a note is played

Regards, Andras

**Comment by jangler** [comment]
I haven't used this feature in other trackers (SunVox I guess) where it exists, but I did find it nice in drum machine sequencing and can imagine it being useful.

My main issue with this is that unlike OpenMPT, which has its own module file format for extensions like this, Schism only works with IT files, and IT files that use this extension would not play back or possibly even load correctly in anything but Schism, and IMO kind of stop even being proper IT files at that point. Maybe there are already features that fit that description which I'm not thinking of, but even if so I wouldn't want to add more.

**Comment by firolightfog** [comment]
I kind of understand your approach. And I can tell that I consider SchismTracker a true gem of legacy software as it is!

On the other hand I don't use it as a traditional tracker. I look at it more like a groove box with limited saving capacities. E.g. I have a bunch of S3M files with AdLib noises only on the F3 page. I regularly open them and set the Instruments options on F12. Later I play with the values on F4 although I know that I won't be able to save it. (I.e. S3m doesn't like instruments, IT doesn like AdLib.) But at least I have some fun.  Specially if I run it through a guitar pedal... :-D

Honestly speaking first I tried to hack the way the Volume Swing value is handled. I wanted to introduce a new parameter (VolumeSwing) to the config file. 0 would mean the regular 'Volume Swing' and setting it 1 would interpret the RV value (aka 'Random volume variation' of ITTECH.txt) as a 'Probability'. It's not a great loss as I can emulate the random volume anyway by Cx, Dx, DFx, DxF, D0x, Dx0, as well as the volume column (see Ctrl-U). Regardless having all those nice files on GitHub I failed miserably due to my limited programming skills. {:^]

Then I thought to open a new topic here hoping a more convenient solution: finding a clever and bored programmer and having a slider for each instruments!

Cheers, Andras
